opsdroid/helper.py

```diff
@@ -44,6 +44,67 @@ def del_rw(action, name, exc):
     os.remove(name)
 
 
+# This is meant to provide backwards compatibility for versions
+# prior to  0.16.0 in the future this will be deleted
+
+
+def convert_dictionary(modules):
+    """Convert dictionary to new format.
+
+    We iterate over all the modules in the list and change the dictionary
+    to be in the format 'name_of_module: { config_params}'
+
+    Args:
+        modules (list): List of dictionaries that contain the module configuration
+
+    Return:
+        List: New modified list following the new format.
+
+    """
+    config = dict()
+
+    try:
+        for module in modules:
+            module_copy = module.copy()
+            del module_copy["name"]
+
+            config[module["name"]] = module_copy
+
+        return config
+    except TypeError:
+        return {}
+
+
+def update_config(config):
```

Comment:
I would be tempted to make this more descriptive. We making a breaking change to our config and this is here for backward compatibility. Maybe something like `update_pre_0_17_config_format`.